substrate/frame/revive/src/exec.rs

```diff
@@ -1675,6 +1690,45 @@ where
 		}
 		true
 	}
+
+	/// Calc the limits for a cross contract call.
+	fn calc_limits(&self, call_resources: &CallResources) -> (Weight, BalanceOf<T>) {
+		match call_resources {
+			CallResources::Precise { weight, deposit_limit } =>
+				(*weight, (*deposit_limit).saturated_into()),
+			CallResources::Ethereum(gas_limit) => {
+				// the resources of the subcall are relative to the available resources
+				let available: BalanceOf<T> = self.gas_left().saturated_into();
+				let gas_limit = (*gas_limit).saturated_into::<BalanceOf<T>>().min(available);
+				let ratio = FixedU128::from_rational(
+					gas_limit.saturated_into(),
+					available.max(1u32.into()).saturated_into(),
+				);
+
+				// weight limit is expected to be set to we can just multiply directly
+				let weight_limit = {
+					let weight_left = self.top_frame().nested_gas.gas_left();
+					Weight::from_parts(
+						ratio.saturating_mul_int(weight_left.ref_time()),
+						ratio.saturating_mul_int(weight_left.proof_size()),
+					)
+				};
+
+				// the gas_limit is in unadjusted fee
+				let deposit_limit = {
+					let weight_fee = T::FeeInfo::weight_to_fee(&weight_limit);
```

Comment:
This should also be a `weight_to_fee` calculation where we take the minimum instead of maximum to be consistent with https://github.com/paritytech/polkadot-sdk/pull/10018/files#r2435295902